pkg/cache/context_recovery_test.go

```diff
@@ -269,15 +269,23 @@ func TestAppRecovery(t *testing.T) {
 	podLister.AddPod(&pod2)
 	podLister.AddPod(&pod3)
```

Comment:
We have three pods, and three apps: if the pod/app does not impact testing we must remove it. If it is crucial for the test we need to check that it is created (or not) etc.

Reply:
Make sense, I'll add validations for apps at last, pods in recovery process have not been handled yet so there's no check for them.

Reply:
Nothing happens with pod 3, we do not check that nor can we so we should not have it in the code.
For pod1 and pod2: if one fails both fail. I would be highly surprised if that did not happen. Having 2 in the code thus does not add anything to the test.

I would really simplify the test to just have one app and make sure it works.
If you want to test the non recovery (i.e. the app3 case) create a new test for that and check that the app never gets to a recovering state.

Reply:
> I would really simplify the test to just have one app and make sure it works.

Make sense to me, I will update it later.

> If you want to test the non recovery (i.e. the app3 case) create a new test for that and check that the app never gets to a recovering state.

How about to call `context.GetApplication("app2")` and make sure that it will return error("application app3 is not found in context") at last of this test? I think this can prove that nothing happens with pod3.